### PR TITLE
Added Contributors Guide to docs folder

### PR DESCRIPTION
**Pull request by kannankvs**
Signed-off-by: kannankvs <kannan_kvs@dell.com>

##### ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- New Module Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by kannankvs** [comment]
Was closed by mistake. Re opening it again

File: docs/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+
+### Feature request
+
+If you have an idea of how to improve this project, submit a [feature request](https://github.com/dell/<<repo>>/issues/new?labels=type%2Ffeature-request%2C+needs-triage&template=feature_request.md&title=%5BFEATURE%5D%3A).
+
+### Answering questions
+
+If you have a question and you can't find the answer in the documentation or issues, the next step is to submit a [question.](https://github.com/dell/<<repo>>/issues/new?labels=type%2Fquestion&template=ask-a-question.md&title=%5BQUESTION%5D%3A)
+
+We'd love your help answering questions being asked by other CSM users.
```
Comment (stalabi1) [inline docs/CONTRIBUTING.md:86]: Changing 'being' to 'that are' reads more clear to me.

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+
+## Triage issues
+
+Triage helps ensure that issues resolve quickly by:
+
+* Ensuring the issue's intent and purpose is conveyed precisely. This is necessary because it can be difficult for an issue to explain how an end user experiences a problem and what actions they took.
+* Giving a contributor the information they need before they commit to resolving an issue.
+authors* Lowering the issue count by preventing duplicate issues.
```
Comment (stalabi1) [inline docs/CONTRIBUTING.md:94]: There seems to be a typo here with 'authors'. I assume 'Lowering the issue count by preventing duplicate issues.' should be its own bullet? It's not showing that way in the md previewer.

File: docs/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+* The pull request closes one related issue.
+* The pull request contains necessary tests that verify the intended behavior.
+* If your pull request has conflicts, rebase your branch onto the main branch.
+
+If the pull request fixes a bug:
+
+* The pull request description must include `Fixes #<issue number>`.
+* To avoid regressions, the pull request should include tests that replicate the fixed bug.
+
+The owning team _squashes_ all commits into one when we accept a pull request. The title of the pull request becomes the subject line of the squashed commit message. We still encourage contributors to write informative commit messages, as they becomes a part of the Git commit body.
```
Comment (stalabi1) [inline docs/CONTRIBUTING.md:167]: There's a typo in the 3rd sentence, 'becomes' should be 'become'.

File: docs/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+
+We use the pull request title when we generate change logs for releases. As such, we strive to make the title as informative as possible.
+
+Make sure that the title for your pull request uses the same format as the subject line in the commit message.
+
+### Quality Gates for pull requests
+
+GitHub Actions are used to enforce quality gates when a pull request is created or when any commit is made to the pull request. These GitHub Actions enforce our minimum code quality requirement for any code that get checked into the code repository. If any of the quality gates fail, it is expected that the contributor will look into the check log, understand the problem and resolve the issue. If help is needed, please feel free to reach out the maintainers of the project for [support](SUPPORT.md).
```
Comment (stalabi1) [inline docs/CONTRIBUTING.md:175]: There are some typos:

-  'any code that get' should be 'any code that **gets**'
-  'reach out the' should be 'reach out **to** the'

**Comment by kannankvs** [comment]
We are unable to commit in this PR due to "head repository deletion" issue. Hence we raised a new [PR#165](https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/165) and copied all the comments from this PR to the new PR. This PR is no more valid and hence we are closing it.

**Pull request closed without merging**